src/target/riscv/riscv_semihosting.c

```diff
@@ -72,33 +115,19 @@ enum semihosting_result riscv_semihosting(struct target *target, int *retval)
 	if (result != ERROR_OK)
 		return SEMIHOSTING_ERROR;
 
-	/*
-	 * The instructions that trigger a semihosting call,
-	 * always uncompressed, should look like:
-	 */
-	uint32_t magic[] = {
-		0x01f01013,	/* slli    zero,zero,0x1f */
-		0x00100073,	/* ebreak */
-		0x40705013	/* srai    zero,zero,0x7 */
-	};
+	bool sequence_found;
+	*retval = riscv_semihosting_detect_magic_sequence(target, pc, &sequence_found);
+	if (*retval != ERROR_OK)
+		return SEMIHOSTING_ERROR;
 
-	/* Read three uncompressed instructions: The previous, the current one (pointed to by PC) and the next one */
-	for (int i = 0; i < 3; i++) {
-		uint8_t buf[4];
-		/* Instruction memories may not support arbitrary read size. Use any size that will work. */
-		target_addr_t address = (pc - 4) + 4 * i;
-		*retval = riscv_read_by_any_size(target, address, 4, buf);
-		if (*retval != ERROR_OK)
-			return SEMIHOSTING_ERROR;
-		uint32_t value = target_buffer_get_u32(target, buf);
-		LOG_TARGET_DEBUG(target, "compare 0x%08x from 0x%" PRIx64 " against 0x%08x",
-			value, address, magic[i]);
-		if (value != magic[i]) {
-			LOG_TARGET_DEBUG(target, "   -> NONE (no magic)");
-			return SEMIHOSTING_NONE;
-		}
+	if (!sequence_found) {
+		LOG_TARGET_DEBUG(target, "   -> NONE (no magic)");
```

Comment:
maybe move this printout inside riscv_semihosting_detect_magic_sequence?

```
		if (result != ERROR_OK) {
			*sequence_found = false;
			return result;
		}
```

Having context-dependent printout like this is somewhat confusing.

Reply:
Addressed, but differently.

My understanding is that the prints like `   -> NONE (no magic)` were supposed to print the final outcome of `riscv_semihosting()` function into the debug log for easier troubleshooting.

I agree that the prints looked confusing.

For consistency and clarity, I changed all of them to this form `Semihosting outcome: SOMETHING (extra explanation)`. Please, let me know if this form looks OK to you.

Reply:
it looks good, thank you!